netket/jax/_batch_utils.py

```diff
@@ -0,0 +1,47 @@
+from netket.jax import compose
+
+
+def unbatch(x):
+    return x.reshape((-1,) + x.shape[2:])
+
+
+def batch(x, batchsize=None):
+    # batchsize=None -> add just a dummy batch dimension, same as np.expand_dims(x, 0)
+    n = x.shape[0]
+    if batchsize is None:
+        batchsize = n
+    else:
+        assert (n % batchsize) == 0
+    return x.reshape((n // batchsize, batchsize) + x.shape[1:])
```

Comment:
can you address giuseppe's comment in the other PR? 
Make batch return a tuple, where the first is the batched data and the second is a function to unbatch.